### PR TITLE
📖 Update jobs documentation

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR updates jobs documentation according to changes in flight

### CODE REVIEW

File: docs/book/src/reference/jobs.md [hunk 1]
```diff
@@ -69,15 +71,16 @@ GitHub (On Release) Workflows:
 [pull-cluster-api-apidiff-main]: https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api#capi-pr-apidiff-main
 [pull-cluster-api-verify]: https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api#capi-pr-verify-main
 [pull-cluster-api-test-main]: https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api#capi-pr-test-main
-[pull-cluster-api-test-main-mink8s]: https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api#capi-pr-test-main-mink8s
+[pull-cluster-api-test-mink8s-main]: https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api#capi-pr-test-mink8s-main
 [pull-cluster-api-e2e-main]: https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api#capi-pr-e2e-main
+[periodic-cluster-api-e2e-v1alpha3-v1beta1-main]: https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api#capi-e2e-v1alpha3-v1beta1-upgrade-main
```
Comment (sbueringer) [inline docs/book/src/reference/jobs.md:76]: nit: maybe move it down to the other periodic jobs

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Comment by fabriziopandini** [comment]
/hold cancel
PRs in test infra are merged

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Pull request merged**